textalyze.rb

```diff
@@ -23,82 +23,91 @@
+  words_array = words.downcase.scan(/\w+/)  #trying scan
+  return words_array  #Returns words as an array. Not sure if I still need letters.
 end
 
 def read_file(filename)
-  begin
-    file_contents = File.read(filename)
-  rescue
-    puts "Error opening file"
-  end
+  #This method reads the file
+  file_contents = File.read(filename)
   return file_contents
 end
 
-def freq_stat(hash)
-  total = 0.0
-  freq_hash = Hash.new(0)
-  hash.each do |k, v|
-    total += v
+def freq_stat(words_hash)
+  total_words = 0.0
+  percent_hash = Hash.new(0)
+  words_hash.each do |word, frequency|
+    total_words += frequency
   end
-  hash.each do |k, v|  #seems like extra work should come back to refactor
-    freq_hash[k] = v/total
+  words_hash.each do |word, frequency|
+    percent_hash[word] = '%.2f' % [((frequency/total_words)*100).round(2)]
   end
-  return freq_hash
+  return percent_hash
 end
 
-def histogram(hash)
+def histogram(words_percentage)
+  #Filler character
   sym = "="
-  hash.each do |k, v|
-    num_sym = sym * (v * 10).to_i
-    puts "#{k} #{v} #{num_sym}"
+  #Iterate over the hash and print the word and percent to the screen.
+  #The word frequency appears a lot smaller might adjust bucket for filler.
+  words_percentage.each do |word, percent|
+    num_sym = sym * (percent * 10).to_i
+    puts "#{word} [ #{percent}%]   #{num_sym}"
   end
 end
 
+def url_download(url = "http://www.gutenberg.org/files/2701/2701-h/2701-h.htm")
+  page = Nokogiri::HTML(open(url))
+  paragraphs = page.css('p')
+  return paragraphs.text
+end
+
 def prints_data()
-  if ARGV.size == 0
-    puts "Input a filename!. Try again!"
-    puts "ruby textalyze.rb yourfile.txt"
-  end
-  begin
-    file_contents = read_file(ARGV[0])
-    #arry = string_split(file_contents)
-    arry = sanitize(file_contents)
-    results = freq_stat(item_counts(arry))
-    sym = "="
-    results.each do |k, v|
-      percent = '%.2f' % [(v*100).round(2)]
-      num_sym = sym * (percent * 10).to_i
-      puts "#{k} [ #{percent}%]   #{num_sym}"
-    end
-  rescue
-    puts "Trouble in paradise"
-  end
+  #struggling a bit with what to name things is there a practice or rule to try and follow?
+  #I was a little confused how to handle words and letters or if that was needed.
+  # if ARGV.size == 0
+  #   puts "Input a filename!. Try again!"
+  #   puts "ruby textalyze.rb yourfile.txt"
+  # end
+  # #Read file specified in ARGV as a sting
+  # file_contents = read_file(ARGV[0])
+  #arry = string_split(file_contents)
+  web_contents = url_download
+  #normalizes words and creates an array of words
+  words_array = words_array(web_contents)
+  #Counts the array of words and finds the frequency percentage that the word is used.
+  #Something isn't right my stats are all zero. Need to come back and fix.
+  words_percentage = freq_stat(item_counts(words_array))
+  #Calls the histogram method and prints the output with a filler character
+  print_histogram = histogram(words_percentage)
```

Comment:
It's not necessary to assign every return value to a variable.  In a case where the method's behavior isn't encoded in the return value but is instead in some side-effect, e.g., printing out text, writing to a file, you can just call it.